packages/context/src/notification/NotificationProvider.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useState, createContext, useEffect, useContext } from 'react';
+import { HttpTransportType, HubConnection, HubConnectionBuilder, LogLevel } from '@microsoft/signalr/dist/esm/index.js';
+import { useApiServices } from '../api/index.js';
+
+export type NotificationConnectionInfo = {
+    url: string;
+    accessToken: string;
+};
+
+export type Subscription<T> = {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
Why does the callback return `any`?  Do we do anything with that returned value?

Reply:
we are not doing anything. this can return void

Reply:
Does it do anything with that return value?

Reply:
behind the scene, it is used the return value to set the complete message (probably logging to their system).

Reply:
What do you mean by "to set the complete message"?

Reply:
their server is expecting a response so they use the callback returned value to construct a completion message.

Reply:
But is that only if the server requests a response?  I.e. do we have to do something on the server side for it to expect the response, otherwise it's just ignored?

Reply:
https://github.com/dotnet/aspnetcore/blob/f545e80531c098e7c1611eca861b0f080fa14a3c/src/SignalR/clients/ts/signalr/src/HubConnection.ts#L756C1-L756C1

This is their implementation. I don't think we need to do anything.

Reply:
```
// Server expects a response
const expectsResponse = invocationMessage.invocationId ? true : false;
```

This suggests to me they don't always expect a response.  In any case, we certainly don't expect a response, so for users of the evoke sdk, this should just be a `void` return.  If signalr wants a response, that's an implementation detail that would be handled by the sdk.

Reply:
got it.